crates/network/src/lib.rs

```diff
@@ -36,6 +36,16 @@ pub trait ReceiptResponse {
     fn contract_address(&self) -> Option<Address>;
 }
 
+/// Transaction Response
+///
+/// This is distinct from [`Transaction`], since this is a JSON-RPC response.
+///
+/// [`Transaction`]: alloy_consensus::Transaction
+pub trait TransactionResponse {
+    /// Hash of the transaction
+    fn hash(&self) -> B256;
```

Comment:
while we're talking renaming in the other issue, should this be `tx_hash` or something similar? will `hash` cause collisions with `Hash::hash`

Reply:
Supportive of `tx_hash` since `PendingTransactionBuilder` also uses `tx_hash`